Unable to deploy portlet plugins

dotCMS trunk is unable to load plugins that include Portlets- this is because the portlet pathing has changed somehow.

Our customers have a number of plugins that also include portlets and this would have broken their screens.


### Update

The error message initially reported is probably over logging by the jersey framework caused by the servlet engine being restarted.   It has no impact on the running of the system,  this may go away if we update jersey to use CDI.

The cause of the issue was found to be the the fact that DotPortlet had the initParams field added so it could be annotated with JAXB annotations and DotPortlet extends Portlet that has the same field.  In this case the object has two versions of the field one on the parent class and one on the super class.   The effect of this and the design of how DotPortlet is being used and serialized meant that when a portlet is installed, its jsp-view init param should be updated with the path to where the plugin files are installed.   The object was being updated but due to the shadowing when the update was being made to the Database the original old value was being used.   When trying to load the CDN portlet page  for example the jsp files cannot be located due to the path being incorrect.

##
The way that JAXB was being used here was fragile and confusing between the use of Portlet and DotPortlet specifically in the way it was extending the main Portlet object.   Instead the main XML serialization objects have been refactored using Jackson and Immutables framework to be self contained classes that can self serialize and deserialize as well as providing a builder to allow mapping between Portlet and DotPortlet without a direct inheritance that could be liable for change.    Even though we have used immutables with Jackson already in the system for json mapping,  this refactor provides a good self-contained example of using it for XML as well ensuring we have separation of these DTO type objects from the functionality they serve.   An interface has been used on the Immutable and its builder to provide a generic mechanism to provide .toXML and fromXML methods directly on the object class, that provide an abstraction layer.    One area we could later consider if we need more mappings between these objects and others e.g. DotPortlet <-> Portlet is to add a mapping framework like MapStruct,  this is especially helpful if we need to modify and support different versions of XML as the internal code changes.  

In changing DotPortlet, DotPortletList and InitParam types we can clean up the underlying PortletAPIImpl and PortletFactoryImpl and ensure that it is both thread-safe and more clear on when converting between Portlet and DotPortlet without implicit casting 




### Steps to Reproduce

Try to install the dotCDN plugin into dotCMS.  The portlet will fail to load.

### Acceptance Criteria

We should be able to get to the portlet.

### dotCMS Version

24.07.x

### Proposed Objective

Application Performance

### Proposed Priority

Priority 1 - Show Stopper

```
[18/07/24 17:48:21:855 GMT]  WARN struts.MultiMessageResources: Duplicate resource property definition (key=was ==> is now): locale.id.hint=Locale is determined by the Language used, or generated by combining Language and Country 2-letter ISO codes. ==> Locales should generally be concise. Many sites use ISO codes, such as 'en' for English or 'es-cr' for Español in Costa Rica.
[18/07/24 17:48:21:869 GMT]  INFO osgi.Activator: Added actionlet: dotCDN Purge
[18/07/24 17:48:21:869 GMT]  WARN mapper.RuntimeExceptionMapper: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down
java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down
	at org.jvnet.hk2.internal.ServiceLocatorImpl.checkState(ServiceLocatorImpl.java:2270) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetInjecteeDescriptor(ServiceLocatorImpl.java:545) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getInjecteeDescriptor(ServiceLocatorImpl.java:594) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.IterableProviderImpl.justInTime(IterableProviderImpl.java:99) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.IterableProviderImpl.get(IterableProviderImpl.java:108) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.glassfish.jersey.internal.JaxrsProviders.getContextResolver(JaxrsProviders.java:111) ~[jersey-common-2.22.1.jar:?]
	at com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider._locateMapperViaProvider(JacksonJsonProvider.java:198) ~[jackson-jaxrs-json-provider-2.16.1.jar:2.16.1]
	at com.fasterxml.jackson.jaxrs.base.ProviderBase.locateMapper(ProviderBase.java:910) ~[jackson-jaxrs-base-2.16.1.jar:2.16.1]
	at com.fasterxml.jackson.jaxrs.base.ProviderBase._endpointForWriting(ProviderBase.java:685) ~[jackson-jaxrs-base-2.16.1.jar:2.16.1]
	at com.fasterxml.jackson.jaxrs.base.ProviderBase.writeTo(ProviderBase.java:560) ~[jackson-jaxrs-base-2.16.1.jar:2.16.1]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:265) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:250) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.internal.JsonWithPaddingInterceptor.aroundWriteTo(JsonWithPaddingInterceptor.java:106) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundWriteTo(MappableExceptionWrapperInterceptor.java:86) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1130) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$Responder.writeResponse(ServerRuntime.java:711) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:444) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:434) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:329) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:471) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:425) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:383) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:336) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:223) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-websocket.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:181) ~[classes/:?]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:61) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.visitor.filter.servlet.VisitorFilter.doFilter(VisitorFilter.java:81) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.vanityurl.filters.VanityURLFilter.doFilter(VanityURLFilter.java:107) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:137) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:53) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:99) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:129) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.NormalizationFilter.doFilter(NormalizationFilter.java:89) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[catalina.jar:9.0.85]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130) ~[catalina.jar:9.0.85]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670) ~[catalina.jar:9.0.85]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[catalina.jar:9.0.85]
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:765) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[catalina.jar:9.0.85]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) ~[catalina.jar:9.0.85]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1732) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:1295) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint.setSocketOptions(Nio2Endpoint.java:339) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint$Nio2Acceptor.completed(Nio2Endpoint.java:479) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint$Nio2Acceptor.completed(Nio2Endpoint.java:415) ~[tomcat-coyote.jar:9.0.85]
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:127) ~[?:?]
	at sun.nio.ch.Invoker.invokeDirect(Invoker.java:158) ~[?:?]
	at sun.nio.ch.Invoker.invoke(Invoker.java:186) ~[?:?]
	at sun.nio.ch.UnixAsynchronousServerSocketChannelImpl.onEvent(UnixAsynchronousServerSocketChannelImpl.java:193) ~[?:?]
	at sun.nio.ch.EPollPort$EventHandlerTask.run(EPollPort.java:306) ~[?:?]
	at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
[18/07/24 17:48:21:871 GMT] ERROR servlet.VisitorFilter: A MultiException has 1 exceptions.  They are:
1. java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down

javax.servlet.ServletException: A MultiException has 1 exceptions.  They are:
1. java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down

	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:487) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:425) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:383) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:336) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:223) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-websocket.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:181) ~[classes/:?]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:61) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.visitor.filter.servlet.VisitorFilter.doFilter(VisitorFilter.java:81) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.vanityurl.filters.VanityURLFilter.doFilter(VanityURLFilter.java:107) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:137) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:53) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:99) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:129) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.NormalizationFilter.doFilter(NormalizationFilter.java:89) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[catalina.jar:9.0.85]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130) ~[catalina.jar:9.0.85]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670) ~[catalina.jar:9.0.85]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[catalina.jar:9.0.85]
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:765) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[catalina.jar:9.0.85]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) ~[catalina.jar:9.0.85]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1732) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:1295) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint.setSocketOptions(Nio2Endpoint.java:339) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint$Nio2Acceptor.completed(Nio2Endpoint.java:479) ~[tomcat-coyote.jar:9.0.85]
	at org.apache.tomcat.util.net.Nio2Endpoint$Nio2Acceptor.completed(Nio2Endpoint.java:415) ~[tomcat-coyote.jar:9.0.85]
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:127) ~[?:?]
	at sun.nio.ch.Invoker.invokeDirect(Invoker.java:158) ~[?:?]
	at sun.nio.ch.Invoker.invoke(Invoker.java:186) ~[?:?]
	at sun.nio.ch.UnixAsynchronousServerSocketChannelImpl.onEvent(UnixAsynchronousServerSocketChannelImpl.java:193) ~[?:?]
	at sun.nio.ch.EPollPort$EventHandlerTask.run(EPollPort.java:306) ~[?:?]
	at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.glassfish.hk2.api.MultiException: A MultiException has 1 exceptions.  They are:
1. java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down

	at org.jvnet.hk2.internal.FactoryCreator.getFactoryHandle(FactoryCreator.java:106) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.FactoryCreator.dispose(FactoryCreator.java:173) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.SystemDescriptor.dispose(SystemDescriptor.java:526) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope$Instance.remove(RequestScope.java:532) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope$Instance.release(RequestScope.java:549) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:319) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:471) ~[jersey-container-servlet-core-2.22.1.jar:?]
	... 79 more
Caused by: java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_46,47,1198896597) has been shut down
	at org.jvnet.hk2.internal.ServiceLocatorImpl.checkState(ServiceLocatorImpl.java:2270) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getServiceHandleImpl(ServiceLocatorImpl.java:639) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getServiceHandle(ServiceLocatorImpl.java:632) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getServiceHandle(ServiceLocatorImpl.java:650) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.FactoryCreator.getFactoryHandle(FactoryCreator.java:103) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.FactoryCreator.dispose(FactoryCreator.java:173) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.SystemDescriptor.dispose(SystemDescriptor.java:526) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope$Instance.remove(RequestScope.java:532) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope$Instance.release(RequestScope.java:549) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:319) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:471) ~[jersey-container-servlet-core-2.22.1.jar:?]
	... 79 more
[18/07/24 17:48:21:872 GMT]  WARN error.custom_002derror_002dpage_jsp: Request failed. on /html/error/custom-error-page.jsp
[18/07/24 17:48:26:972 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:26:972 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:26:976 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:26:977 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:27:001 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:27:001 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:27:004 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:48:27:008 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:50:41:598 GMT]  INFO maintenance.MaintenanceResource: User 'Will Ezell [ID: dotcms.org.5][email:will@dotcms.com]' is shutting down dotCMS Cluster with a rolling delay of 60
[18/07/24 17:50:41:598 GMT]  INFO util.SecurityLogger: class com.dotcms.rest.api.v1.maintenance.MaintenanceResource : User 'Will Ezell [ID: dotcms.org.5][email:will@dotcms.com]' is shutting down dotCMS Cluster with a rolling delay of 60 -- ip:209.134.33.209,user:Will Ezell [ID: dotcms.org.5][email:will@dotcms.com]
[18/07/24 17:50:41:714 GMT]  INFO maintenance.ClusterManagementTopic: Got DOTCMS_CLUSTER_RESTART from server:5b1044a37d.
[18/07/24 17:50:41:717 GMT]  INFO maintenance.ClusterManagementTopic: Restarting  2024-07-18T17:50:41.715490Zs, serverOrder:0, rollingDelay:60
[18/07/24 17:50:41:727 GMT]  INFO maintenance.ClusterManagementTopic: Running restart in: 0 seconds
[18/07/24 17:50:41:728 GMT]  INFO job.ShutdownHookThread: Running dotCMS shutdown cleanup sequence.
[18/07/24 17:50:41:744 GMT]  INFO core.QuartzScheduler: Scheduler dotCMSQuartz_$_NON_CLUSTERED started.
[18/07/24 17:50:41:744 GMT]  INFO core.QuartzScheduler: Scheduler dotCMSQuartz_$_NON_CLUSTERED shutting down.
[18/07/24 17:50:41:744 GMT]  INFO core.QuartzScheduler: Scheduler dotCMSQuartz_$_NON_CLUSTERED paused.
[18/07/24 17:50:41:753 GMT]  INFO core.QuartzScheduler: Scheduler dotCMSQuartz_$_NON_CLUSTERED shutdown complete.
[18/07/24 17:50:41:764 GMT]  INFO servlets.InitServlet: dotCMS shutting down OSGI
[18/07/24 17:50:41:777 GMT]  WARN reindex.ReindexThread: ---  ReindexThread is stopping, background indexing will not take place
[18/07/24 17:50:41:790 GMT]  INFO job.Activator: Stopping Content Unlock Job
[18/07/24 17:50:41:790 GMT]  INFO rest.Activator: Removing new Restful OSGI INFO Service:OSGIInfoResource
[18/07/24 17:50:41:791 GMT]  INFO encoder.Activator: Removed View Tool: owasp
[18/07/24 17:50:41:791 GMT]  INFO flexmark.Activator: Removed View Tool: flexmark
[18/07/24 17:50:41:791 GMT]  INFO osgi.Activator: Stopping dotZapier Plugin
[18/07/24 17:50:41:791 GMT]  INFO osgi.Activator: Removed View Tool: zapier
[18/07/24 17:50:41:792 GMT]  INFO osgi.Activator: Removed View Tool: dotcdn
[18/07/24 17:50:41:805 GMT]  INFO servlets.InitServlet: dotCMS shutting down Elastic Search
[18/07/24 17:50:41:805 GMT]  INFO servlets.InitServlet: dotCMS shutting down
[18/07/24 17:50:41:815 GMT]  INFO filters.CharsetEncodingFilter: Destroying character encoding filter...
[18/07/24 17:50:41:816 GMT]  INFO servlet.VisitorFilter: VisitorLogger Filter Destroyed
[18/07/24 17:50:41:912 GMT]  INFO listeners.ContextLifecycleListener: Shutdown : Started, executing a clean shutdown.
[18/07/24 17:50:41:915 GMT]  INFO enterprise.LicenseUtil: Setting up default license
[18/07/24 17:50:41:917 GMT]  INFO timedcache.TimedCacheProvider: ===== Calling shutdown [Timed Cache Provider].
[18/07/24 17:50:41:917 GMT]  INFO caffine.CaffineCache: ===== Calling shutdown [Caffine Memory Cache].
[18/07/24 17:50:41:917 GMT]  INFO h22.H22Cache: Building new H22 Cache, db:0
[18/07/24 17:50:41:917 GMT]  INFO h22.H22HikariPool: H22 on disk cache:jdbc:h2:/data/local/dotsecure/h22cache/0/2024-07-18_13-50-41/cache;DB_CLOSE_ON_EXIT=FALSE
[18/07/24 17:50:41:931 GMT]  INFO h22.H22Cache: Building new H22 Cache, db:1
[18/07/24 17:50:41:932 GMT]  INFO h22.H22HikariPool: H22 on disk cache:jdbc:h2:/data/local/dotsecure/h22cache/1/2024-07-18_13-50-41/cache;DB_CLOSE_ON_EXIT=FALSE
[18/07/24 17:50:41:946 GMT]  INFO listeners.ContextLifecycleListener: Shutdown : Finished.
command terminated with exit code 137
(base) ➜  plugin-dotcms-license git:(main) ✗ k exec -c dotcms -it dotcms-corpsites-auth-0 -n corpsites bash
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
dotcms@dotcms-corpsites-auth-0:/$ tail -f /srv/
00-config-defaults.sh        30-override-config-props.sh  50-load-dump-sql.sh          config/                      entrypoint.sh                templates/
20-copy-overriden-files.sh   40-custom-starter-zip.sh     OVERRIDE/                    dotserver/                   home/                        utils/
dotcms@dotcms-corpsites-auth-0:/$ tail -f /srv/dotserver/tomcat/
.cargo           CONTRIBUTING.md  NOTICE           RELEASE-NOTES    bin/             conf/            logs/            temp/            work/
BUILDING.txt     LICENSE          README.md        RUNNING.txt      common/          lib/             shared/          webapps/
dotcms@dotcms-corpsites-auth-0:/$ tail -f /srv/dotserver/tomcat/logs/
catalina.2024-07-18.log       dotcms.log                    dotcms_access.2024-07-18.log
dotcms@dotcms-corpsites-auth-0:/$ tail -f /srv/dotserver/tomcat/logs/dotcms.log
	at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$2.completed(Nio2Endpoint.java:619) ~[tomcat-coyote.jar:9.0.85]
	at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:127) ~[?:?]
	at sun.nio.ch.Invoker$2.run(Invoker.java:219) ~[?:?]
	at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
[18/07/24 17:53:01:606 GMT]  INFO services.DotResourceCache: velocityMiss:VelocityResourceKey [path=/ext/license.vtl, language=1, id1=license, id2=null, type=VTL, mode=LIVE] @ url:GET//dotcms.com/c/portal/layout | lang:1 | ip:70.50.106.46 | Admin:true | start:07-18-2024 05:51:36 GMT  ref:https://authoring.dotcms.com/dotAdmin/  ?p_l_id=31d82c3e-b796-41c5-8e3d-63a62ac00480&p_p_id=EXT_DOTCMS_LICENSE_PORTLET&p_p_action=0&&dm_rlout=1&r=1721325176231&in_frame=true&frame=detailFrame&container=true&angularCurrentPortlet=EXT_DOTCMS_LICENSE_PORTLET
[18/07/24 17:53:01:606 GMT]  WARN resource.ResourceManagerImpl: ResourceManager : unable to find resource '/ext/license.vtl' in any resource loader. @ url:GET//dotcms.com/c/portal/layout | lang:1 | ip:70.50.106.46 | Admin:true | start:07-18-2024 05:51:36 GMT  ref:https://authoring.dotcms.com/dotAdmin/  ?p_l_id=31d82c3e-b796-41c5-8e3d-63a62ac00480&p_p_id=EXT_DOTCMS_LICENSE_PORTLET&p_p_action=0&&dm_rlout=1&r=1721325176231&in_frame=true&frame=detailFrame&container=true&angularCurrentPortlet=EXT_DOTCMS_LICENSE_PORTLET
[18/07/24 17:53:01:611 GMT] ERROR servlet.MainServlet: com.dotcms.repackage.javax.portlet.PortletException
[18/07/24 17:53:05:701 GMT]  INFO apps.SecretsKeyStoreHelper: KeyStore loaded successfully after `1` tries.
[18/07/24 17:53:06:452 GMT]  INFO osgi.OSGIResource: Portlets Removed: [EXT_DOTCMS_LICENSE_PORTLET]
[18/07/24 17:53:06:453 GMT]  INFO osgi.OSGIResource: Actionlets Removed: []
[18/07/24 17:53:06:458 GMT]  WARN mapper.RuntimeExceptionMapper: ServiceLocatorImpl(__HK2_Generated_5,6,763315407) has been shut down
java.lang.IllegalStateException: ServiceLocatorImpl(__HK2_Generated_5,6,763315407) has been shut down
	at org.jvnet.hk2.internal.ServiceLocatorImpl.checkState(ServiceLocatorImpl.java:2270) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetInjecteeDescriptor(ServiceLocatorImpl.java:545) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getInjecteeDescriptor(ServiceLocatorImpl.java:594) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.IterableProviderImpl.justInTime(IterableProviderImpl.java:99) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.jvnet.hk2.internal.IterableProviderImpl.get(IterableProviderImpl.java:108) ~[hk2-locator-2.4.0-b31.jar:?]
	at org.glassfish.jersey.internal.JaxrsProviders.getContextResolver(JaxrsProviders.java:111) ~[jersey-common-2.22.1.jar:?]
	at com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider._locateMapperViaProvider(JacksonJsonProvider.java:198) ~[jackson-jaxrs-json-provider-2.16.1.jar:2.16.1]
	at com.fasterxml.jackson.jaxrs.base.ProviderBase.locateMapper(ProviderBase.java:910) ~[jackson-jaxrs-base-2.16.1.jar:2.16.1]
	at com.fasterxml.jackson.jaxrs.base.ProviderBase._endpointForWriting(ProviderBase.java:685) ~[jackson-jaxrs-base-2.16.1.jar:2.16.1]
	at com.fasterxml.jackson.jaxrs.base.ProviderBase.writeTo(ProviderBase.java:560) ~[jackson-jaxrs-base-2.16.1.jar:2.16.1]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:265) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:250) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.internal.JsonWithPaddingInterceptor.aroundWriteTo(JsonWithPaddingInterceptor.java:106) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.internal.MappableExceptionWrapperInterceptor.aroundWriteTo(MappableExceptionWrapperInterceptor.java:86) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1130) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$Responder.writeResponse(ServerRuntime.java:711) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$Responder.processResponse(ServerRuntime.java:444) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:434) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:329) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) ~[jersey-common-2.22.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) ~[jersey-server-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:471) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:425) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:383) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:336) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:223) ~[jersey-container-servlet-core-2.22.1.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-websocket.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:181) ~[classes/:?]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:61) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.visitor.filter.servlet.VisitorFilter.doFilter(VisitorFilter.java:81) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.vanityurl.filters.VanityURLFilter.doFilter(VanityURLFilter.java:107) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[urlrewritefilter-4.0.4.jar:4.0.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:137) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:53) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:99) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:129) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
	at com.dotcms.filters.NormalizationFilter.doFilter(NormalizationFilter.java:89) ~[classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[catalina.jar:9.0.85]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[catalina.jar:9.0.85]
```

--- a/config/user/logging/log4j2.xml
+++ b/config/user/logging/log4j2.xml
@@ -1,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration monitorInterval="10">
+    <Properties>
+        <Property name="DEFAULT_CONSOLE_PATTERN">%d{HH:mm:ss.SSS} hello5 [%X] %highlight{%-5level} [%tn] %c{1.}.%M(%F:%L) - %m%n%ex{filters(${stack.filter})}</Property>
+        <Property name="DOTCMS_LOGGING_INCLUDE_LOCATION">false</Property>
+    </Properties>
+    <Loggers>
+        <Logger name="com.dotcms" level="info" additivity="true">
+        </Logger>
+    </Loggers>
+
+    <Loggers>
+        <Logger name="com.dotcms.concurrent.DotConcurrentFactory" level="info" additivity="true">
+        </Logger>
+    </Loggers>
+    <Loggers>
+        <Logger name="com.dotmarketing.filters.ThreadNameFilter" level="warn" additivity="true">
+        </Logger>
+    </Loggers>
+
+
+</Configuration>

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java
@@ -46,6 +46,9 @@ public class PortletFactoryImpl extends PrincipalBean implements PortletFactory 
 
   private final String[] systemXmlFiles;
 
+
+
+
   /**
    * Creates an instance of this Factory using the specified XML files as sources of Portlet
    * definitions.
@@ -94,73 +97,29 @@ public class PortletFactoryImpl extends PrincipalBean implements PortletFactory 
    *
    * @throws JAXBException An error occurred while mapping the XML file into Java objects.
    */
+  /**
+   * Takes the Input Stream of an XML file and extracts the Portlet definitions from it into
+   * Portlet objects.
+   *
+   * @param fileStream The {@link InputStream} of the XML file to read.
+   *
+   * @return A map with the {@link Portlet} definitions.
+   *
+   * @throws JAXBException An error occurred while mapping the XML file into Java objects.
+   */
   public Map<String, Portlet> xmlToPortlets(InputStream fileStream) throws IOException, ParserConfigurationException, SAXException {
     final Map<String, Portlet> portlets = new HashMap<>();
 
     if (fileStream == null) {
       return portlets;
     }
-    SAXParserFactory factory = SAXParserFactory.newInstance();
-    SAXParser saxParser = factory.newSAXParser();
 
-    DefaultHandler handler = new DefaultHandler() {
-      private final StringBuilder currentValue = new StringBuilder();
-      private String portletName;
-      private String portletClass;
-      private final StringBuilder portletElement = new StringBuilder();
-      private boolean isPortlet = false;
-
-      @Override
-      public void startElement(String uri, String localName, String qName, Attributes attributes) {
-        currentValue.setLength(0); // Clear the current value
-        if (qName.equalsIgnoreCase("portlet")) {
-          portletElement.setLength(0); // Clear the portlet element content
-          isPortlet = true;
-        }
-        if (isPortlet) {
-          portletElement.append("<").append(qName);
-          for (int i = 0; i < attributes.getLength(); i++) {
-            portletElement.append(" ").append(attributes.getQName(i)).append("=\"").append(attributes.getValue(i)).append("\"");
-          }
-          portletElement.append(">");
-        }
-      }
-
-      @Override
-      public void characters(char[] ch, int start, int length) {
-        currentValue.append(ch, start, length);
-        if (isPortlet) {
-          portletElement.append(ch, start, length);
-        }
-      }
-
-      @Override
-      public void endElement(String uri, String localName, String qName) {
-        if (qName.equalsIgnoreCase("portlet-name")) {
-          portletName = currentValue.toString().trim();
-          portletElement.append("</").append(qName).append(">");
-        } else if (qName.equalsIgnoreCase("portlet-class")) {
-          portletClass = currentValue.toString().trim();
-          portletElement.append("</").append(qName).append(">");
-        } else if (qName.equalsIgnoreCase("portlet")) {
-          if (portletName != null && !portletName.isEmpty() && portletClass != null && !portletClass.isEmpty()) {
-            portletElement.append("</").append(qName).append(">");
-            String portletXML = portletElement.toString();
-            xmlToPortlet(portletXML).ifPresent(value -> portlets.put(value.getPortletId(), value));
-          }
-          portletName = null;
-          portletClass = null;
-          isPortlet = false;
-        } else if (isPortlet) {
-          portletElement.append("</").append(qName).append(">");
-        }
-      }
-    };
-
-
+    SAXParser saxParser = ThreadLocalSaxParserFactory.getSaxParser();
+    PortletSaxHandler<Portlet> handler = new PortletSaxHandler<>(this::xmlToPortlet);
     saxParser.parse(fileStream, handler);
-    return portlets;
+    return handler.getValueMap();
   }
+
 
 
   @Override

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java
@@ -26,10 +26,7 @@ import java.io.InputStream;
 import java.util.*;
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.parsers.SAXParser;
-import javax.xml.parsers.SAXParserFactory;
-import org.xml.sax.Attributes;
 import org.xml.sax.SAXException;
-import org.xml.sax.helpers.DefaultHandler;
 
 /**
  * Implementation class for the {@link PortletFactory} interface. This class uses JAXB to map XML

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletFactoryImpl.java
@@ -24,6 +24,12 @@ import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.*;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
 
 /**
  * Implementation class for the {@link PortletFactory} interface. This class uses JAXB to map XML
@@ -68,7 +74,7 @@ public class PortletFactoryImpl extends PrincipalBean implements PortletFactory 
    * @throws IOException   An error occurred while reading the XML file.
    * @throws JAXBException An error occurred while mapping the XML file into Java objects.
    */
-  private Map<String, Portlet> xmlToPortlets(final String pathToXmlFile) throws IOException, JAXBException {
+  private Map<String, Portlet> xmlToPortlets(final String pathToXmlFile) throws IOException, ParserConfigurationException, SAXException {
     if (UtilMethods.isNotSet(pathToXmlFile)) {
       return new HashMap<>();
     }
@@ -88,29 +94,84 @@ public class PortletFactoryImpl extends PrincipalBean implements PortletFactory 
    *
    * @throws JAXBException An error occurred while mapping the XML file into Java objects.
    */
-  private Map<String, Portlet> xmlToPortlets(final InputStream fileStream) throws IOException {
-    if (fileStream == null) {
-      return new HashMap<>();
-    }
+  public Map<String, Portlet> xmlToPortlets(InputStream fileStream) throws IOException, ParserConfigurationException, SAXException {
     final Map<String, Portlet> portlets = new HashMap<>();
-    final PortletList portletList = PortletList.builder().fromXml(fileStream);
-    int counter = 1;
-    if (UtilMethods.isSet(portletList) && UtilMethods.isSet(portletList.getPortlets())) {
-      for (final DotPortlet dotPortlets : portletList.getPortlets()) {
-        portlets.put(dotPortlets.getPortletId(), dotPortlets.toPortlet());
-        Logger.debug(this, String.format("%d. Loading portlet ID '%s'", counter, dotPortlets.getPortletId()));
-        counter++;
-      }
+
+    if (fileStream == null) {
+      return portlets;
     }
+    SAXParserFactory factory = SAXParserFactory.newInstance();
+    SAXParser saxParser = factory.newSAXParser();
+
+    DefaultHandler handler = new DefaultHandler() {
+      private final StringBuilder currentValue = new StringBuilder();
+      private String portletName;
+      private String portletClass;
+      private final StringBuilder portletElement = new StringBuilder();
+      private boolean isPortlet = false;
+
+      @Override
+      public void startElement(String uri, String localName, String qName, Attributes attributes) {
+        currentValue.setLength(0); // Clear the current value
+        if (qName.equalsIgnoreCase("portlet")) {
+          portletElement.setLength(0); // Clear the portlet element content
+          isPortlet = true;
+        }
+        if (isPortlet) {
+          portletElement.append("<").append(qName);
+          for (int i = 0; i < attributes.getLength(); i++) {
+            portletElement.append(" ").append(attributes.getQName(i)).append("=\"").append(attributes.getValue(i)).append("\"");
+          }
+          portletElement.append(">");
+        }
+      }
+
+      @Override
+      public void characters(char[] ch, int start, int length) {
+        currentValue.append(ch, start, length);
+        if (isPortlet) {
+          portletElement.append(ch, start, length);
+        }
+      }
+
+      @Override
+      public void endElement(String uri, String localName, String qName) {
+        if (qName.equalsIgnoreCase("portlet-name")) {
+          portletName = currentValue.toString().trim();
+          portletElement.append("</").append(qName).append(">");
+        } else if (qName.equalsIgnoreCase("portlet-class")) {
+          portletClass = currentValue.toString().trim();
+          portletElement.append("</").append(qName).append(">");
+        } else if (qName.equalsIgnoreCase("portlet")) {
+          if (portletName != null && !portletName.isEmpty() && portletClass != null && !portletClass.isEmpty()) {
+            portletElement.append("</").append(qName).append(">");
+            String portletXML = portletElement.toString();
+            xmlToPortlet(portletXML).ifPresent(value -> portlets.put(value.getPortletId(), value));
+          }
+          portletName = null;
+          portletClass = null;
+          isPortlet = false;
+        } else if (isPortlet) {
+          portletElement.append("</").append(qName).append(">");
+        }
+      }
+    };
+
+
+    saxParser.parse(fileStream, handler);
     return portlets;
   }
 
+
   @Override
   @VisibleForTesting
-  public Optional<Portlet> xmlToPortlet(final String xml) throws IOException {
-    final DotPortlet portlet = DotPortlet.builder().fromXml(xml);
-    if (portlet.getPortletId() == null || portlet.getPortletClass() == null) {
-      return Optional.empty();
+  public Optional<Portlet> xmlToPortlet(final String xml){
+    final DotPortlet portlet;
+    try {
+        portlet = DotPortlet.builder().fromXml(xml);
+    } catch (IOException e) {
+       Logger.debug(PortletFactoryImpl.class,"Not a DotPortlet skipping "+xml);
+       return Optional.empty();
     }
     return Optional.of(portlet.toPortlet());
   }

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletSaxHandler.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletSaxHandler.java
@@ -1,0 +1,86 @@
+package com.dotmarketing.business.portal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class PortletSaxHandler<T> extends DefaultHandler {
+
+    private final StringBuilder currentValue = new StringBuilder();
+    private final Function<String,Optional<T>> valueMapper;
+    private final Map<String, T> valueMap;
+    private String portletName;
+    private String portletClass;
+    private final StringBuilder portletElement = new StringBuilder();
+    private boolean isPortlet = false;
+
+    PortletSaxHandler(Function<String,Optional<T>>  valueMapper) {
+        this.valueMap = new HashMap<>();
+        this.valueMapper = valueMapper;
+    }
+
+    public Map<String, T> getValueMap() {
+        return valueMap;
+    }
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) {
+        currentValue.setLength(0);
+        if (qName.equalsIgnoreCase("portlet")) {
+            portletElement.setLength(0);
+            isPortlet = true;
+        }
+        if (isPortlet) {
+            appendStartElement(qName, attributes);
+        }
+    }
+
+    @Override
+    public void characters(char[] ch, int start, int length) {
+        currentValue.append(ch, start, length);
+        if (isPortlet) {
+            portletElement.append(ch, start, length);
+        }
+    }
+
+    @Override
+    public void endElement(String uri, String localName, String qName) {
+        handleEndElement(qName);
+    }
+
+    private void appendStartElement(String qName, Attributes attributes) {
+        portletElement.append("<").append(qName);
+        for (int i = 0; i < attributes.getLength(); i++) {
+            portletElement.append(" ").append(attributes.getQName(i)).append("=\"").append(attributes.getValue(i)).append("\"");
+        }
+        portletElement.append(">");
+    }
+
+    private void handleEndElement(String qName) {
+        if (qName.equalsIgnoreCase("portlet-name")) {
+            portletName = currentValue.toString().trim();
+        } else if (qName.equalsIgnoreCase("portlet-class")) {
+            portletClass = currentValue.toString().trim();
+        } else if (qName.equalsIgnoreCase("portlet")) {
+            processPortletEndElement();
+        }
+        if (isPortlet) {
+            portletElement.append("</").append(qName).append(">");
+        }
+    }
+
+    private void processPortletEndElement() {
+        if (portletName != null && !portletName.isEmpty() && portletClass != null && !portletClass.isEmpty()) {
+            String portletXML = portletElement.toString();
+            valueMapper.apply(portletXML).ifPresent(value -> valueMap.put(portletName, value));
+        }
+        portletName = null;
+        portletClass = null;
+        isPortlet = false;
+    }
+
+
+}

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletSaxHandler.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/PortletSaxHandler.java
@@ -10,14 +10,14 @@ import org.xml.sax.helpers.DefaultHandler;
 public class PortletSaxHandler<T> extends DefaultHandler {
 
     private final StringBuilder currentValue = new StringBuilder();
-    private final Function<String,Optional<T>> valueMapper;
+    private final Function<String, Optional<T>> valueMapper;
     private final Map<String, T> valueMap;
     private String portletName;
     private String portletClass;
     private final StringBuilder portletElement = new StringBuilder();
     private boolean isPortlet = false;
 
-    PortletSaxHandler(Function<String,Optional<T>>  valueMapper) {
+    PortletSaxHandler(Function<String, Optional<T>> valueMapper) {
         this.valueMap = new HashMap<>();
         this.valueMapper = valueMapper;
     }
@@ -48,6 +48,9 @@ public class PortletSaxHandler<T> extends DefaultHandler {
 
     @Override
     public void endElement(String uri, String localName, String qName) {
+        if (isPortlet) {
+            portletElement.append("</").append(qName).append(">");
+        }
         handleEndElement(qName);
     }
 
@@ -67,9 +70,6 @@ public class PortletSaxHandler<T> extends DefaultHandler {
         } else if (qName.equalsIgnoreCase("portlet")) {
             processPortletEndElement();
         }
-        if (isPortlet) {
-            portletElement.append("</").append(qName).append(">");
-        }
     }
 
     private void processPortletEndElement() {
@@ -81,6 +81,4 @@ public class PortletSaxHandler<T> extends DefaultHandler {
         portletClass = null;
         isPortlet = false;
     }
-
-
 }

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/ThreadLocalSaxParserFactory.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/ThreadLocalSaxParserFactory.java
@@ -1,6 +1,8 @@
 package com.dotmarketing.business.portal;
 
 import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Logger;
+import javax.xml.XMLConstants;
 import javax.xml.parsers.ParserConfigurationException;
 import javax.xml.parsers.SAXParser;
 import javax.xml.parsers.SAXParserFactory;
@@ -19,9 +21,22 @@ public class ThreadLocalSaxParserFactory {
                             SAXParserFactory factory = SAXParserFactory.newInstance();
                             // https://rules.sonarsource.com/java/RSPEC-2755
                             // prevent XXE, completely disable DOCTYPE declaration:
-                            factory.setFeature(
-                                    "http://apache.org/xml/features/disallow-doctype-decl", true);
+                            // may not be able to use for backwards compatibility
+                            //factory.setFeature(
+                            //        "http://apache.org/xml/features/disallow-doctype-decl", true);
+
+                            factory.setFeature("http://xml.org/sax/features/external-general-entities",false);
+                            factory.setFeature("http://xml.org/sax/features/external-parameter-entities",false);
+                            factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd",false);
+                            factory.setXIncludeAware(false);
+                            factory.setValidating(false);
+                            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+
                             return factory;
+                        } catch (ParserConfigurationException e) {
+                            Logger.error(ThreadLocalSaxParserFactory.class,"ParserConfigurationException was thrown. The feature 'XMLConstants.FEATURE_SECURE_PROCESSING'"
+                                    + " is probably not supported by your XML processor.",e);
+                            throw new DotRuntimeException(e);
                         } catch (Exception e) {
                             throw new DotRuntimeException(e);
                         }

--- a/dotCMS/src/main/java/com/dotmarketing/business/portal/ThreadLocalSaxParserFactory.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/portal/ThreadLocalSaxParserFactory.java
@@ -1,0 +1,38 @@
+package com.dotmarketing.business.portal;
+
+import com.dotmarketing.exception.DotRuntimeException;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import org.xml.sax.SAXException;
+
+public class ThreadLocalSaxParserFactory {
+
+    private ThreadLocalSaxParserFactory() {
+        // utility class, do not instantiate
+    }
+
+    private static final ThreadLocal<SAXParserFactory> saxParserFactoryThreadLocal =
+            ThreadLocal.withInitial(
+                    () -> {
+                        try {
+                            SAXParserFactory factory = SAXParserFactory.newInstance();
+                            // https://rules.sonarsource.com/java/RSPEC-2755
+                            // prevent XXE, completely disable DOCTYPE declaration:
+                            factory.setFeature(
+                                    "http://apache.org/xml/features/disallow-doctype-decl", true);
+                            return factory;
+                        } catch (Exception e) {
+                            throw new DotRuntimeException(e);
+                        }
+                    });
+
+    public static SAXParser getSaxParser() throws ParserConfigurationException, SAXException {
+        // you could catch and re-throw the RuntimeException if the caller should handle it
+        return saxParserFactoryThreadLocal.get().newSAXParser();
+    }
+
+    public static void cleanup() {
+        saxParserFactoryThreadLocal.remove();
+    }
+}

--- a/dotcms-integration/src/test/java/com/dotmarketing/business/portal/PortletAPIImplTest.java
+++ b/dotcms-integration/src/test/java/com/dotmarketing/business/portal/PortletAPIImplTest.java
@@ -23,13 +23,20 @@ import com.liferay.portlet.StrutsPortlet;
 import com.tngtech.java.junit.dataprovider.DataProvider;
 import com.tngtech.java.junit.dataprovider.DataProviderRunner;
 import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
+import javax.xml.parsers.ParserConfigurationException;
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.xml.sax.SAXException;
 
 @RunWith(DataProviderRunner.class)
 public class PortletAPIImplTest {
@@ -73,6 +80,24 @@ public class PortletAPIImplTest {
                 .build();
 
         return portletApi.savePortlet(newPortlet.toPortlet(), systemUser);
+    }
+
+    /**
+     * Test case to ensure no portlets are returned from the mixed-portlets-test.xml file
+     * It should handle without error  but ignore portlet elements within the portlets element
+     * that do not contain a portlet-name and portlet-class
+     */
+    @Test
+    public void test_portletsFromMixedPortletXml() throws Exception {
+        String resourcePath = "/com/dotmarketing/business/portal/mixed-portlets-test.xml";
+        try (InputStream stream = getClass().getResourceAsStream(resourcePath)) {
+            assertNotNull("Resource not found: " + resourcePath, stream);
+            Map<String, Portlet> portlets = new PortletFactoryImpl().xmlToPortlets(stream);
+            assertTrue("Expecting exactly 1 valid portlet",portlets.size() == 1);
+        } catch (IOException | ParserConfigurationException | SAXException e) {
+            Logger.error(this, "Error loading portlets from liferay-portlet.xml", e);
+            Assert.fail("Exception occurred: " + e.getMessage());
+        }
     }
 
     /**

--- a/dotcms-integration/src/test/java/com/dotmarketing/business/portal/PortletAPIImplTest.java
+++ b/dotcms-integration/src/test/java/com/dotmarketing/business/portal/PortletAPIImplTest.java
@@ -93,7 +93,7 @@ public class PortletAPIImplTest {
         try (InputStream stream = getClass().getResourceAsStream(resourcePath)) {
             assertNotNull("Resource not found: " + resourcePath, stream);
             Map<String, Portlet> portlets = new PortletFactoryImpl().xmlToPortlets(stream);
-            assertTrue("Expecting exactly 1 valid portlet",portlets.size() == 1);
+            assertEquals("Expecting exactly 1 valid portlet", 1, portlets.size());
         } catch (IOException | ParserConfigurationException | SAXException e) {
             Logger.error(this, "Error loading portlets from liferay-portlet.xml", e);
             Assert.fail("Exception occurred: " + e.getMessage());

--- a/dotcms-integration/src/test/resources/com/dotmarketing/business/portal/mixed-portlets-test.xml
+++ b/dotcms-integration/src/test/resources/com/dotmarketing/business/portal/mixed-portlets-test.xml
@@ -1,0 +1,19 @@
+<?xml version="1.0"?>
+<!DOCTYPE portlets>
+
+<portlets>
+    <portlet id="EXT_HELLO_WORLD" narrow="true"/>
+    <portlet id="EXT_STRUTS_HELLO_WORLD" struts-path="ext/strutshello" narrow="true"/>
+    <!-- Add one valid portlet to check that others are ignored -->
+    <portlet>
+        <portlet-name>files-legacy</portlet-name>
+        <display-name>File Manager</display-name>
+        <portlet-class>com.liferay.portlet.StrutsPortlet</portlet-class>
+        <init-param>
+            <name>view-action</name>
+            <value>/ext/files/view_files</value>
+        </init-param>
+        <resource-bundle>com.liferay.portlet.StrutsResourceBundle</resource-bundle>
+    </portlet>
+    <portlet id="EXT_JSP_HELLO_WORLD" narrow="true"/>
+</portlets>

--- a/justfile
+++ b/justfile
@@ -59,7 +59,7 @@ build-select-module-deps module=":dotcms-core":
 
 # Starts the dotCMS application in a Docker container on a dynamic port, running in the foreground
 dev-run:
-    ./mvnw -pl :dotcms-core -Pdocker-start
+    ./mvnw -pl :dotcms-core -Pdocker-start,debug-suspend
 
 # Maps paths in the docker container to local paths, useful for development
 dev-run-map-dev-paths:

